### ISSUE TITLE
[Enhancement] DataParallel without trainer

### ISSUE THREAD

**Issue by justusschock** [chainer/chainer#7264]
Hi, 
I've implemented a version of DataParallel, which automatically handles the following things:

* replication of the model across different devices
* data scattering across thoose devices
* the gradient accumulation before optimizer update
* the update of all replicated parameters after the optimizer step

My implementation is based on the [DataParallel Example without a trainer](https://docs.chainer.org/en/stable/guides/gpu.html#data-parallel-computation-on-multiple-gpus-without-trainer) and the [PyTorch Version](https://pytorch.org/docs/stable/nn.html#torch.nn.DataParallel).

Would you welcome a PR including this or is this something you don't want?

**Comment by crcrpar** [comment]
+1 for this and a PR for that will be really helpful even if not merged.
